src/vellum/workflows/inputs/base.py

```diff
@@ -40,14 +42,38 @@ class BaseInputs(metaclass=_BaseInputsMeta):
     __parent_class__: Type = type(None)
 
     def __init__(self, **kwargs: Any) -> None:
+        for name, field_type in self.__class__.__annotations__.items():
+            if name not in kwargs and name not in vars(self.__class__):
```

Comment:
[nit] we already do some of this magic in the metaclass, we should be able to just do `for input_descriptor in self.__class__:` like in the `__iter__` method below